### PR TITLE
fix withClause

### PR DESCRIPTION
**Pull request by czpmango**
Fixes [https://jira.nebula-graph.io/browse/NG-435](https://jira.nebula-graph.io/browse/NG-435)

This PR fixes the bug that the order by operator becomes invalid due to unstable filtering in the filter executor.

### CODE REVIEW

File: tests/tck/features/match/WithUnwind.feature [hunk 1]
```diff
@@ -186,3 +186,36 @@ Feature: With clause and Unwind clause
...
       | "hello" | 3 | 3 |
+
+  Scenario: match with return
+    When executing query:
+      """
+      MATCH (v:player)
+      WITH v.age AS age, v AS v, v.name AS name
+         ORDER BY age DESCENDING, name ASCENDING
+         LIMIT 20
+         WHERE age > 30
```
Comment (Shylock-Hg) [inline tests/tck/features/match/WithUnwind.feature:197]: Is there order requirement of order by and where ? I think filter first then sort is equivalent.

Comment (czpmango) [reply tests/tck/features/match/WithUnwind.feature:197]: The where clause in the with clause must be executed last, which is specified by the openCypher standard. In fact, the where clause here is more like the having clause in sql.
You can refer to [https://neo4j.com/docs/cypher-manual/current/clauses/with/#with-filter-on-aggregate-function-results](https://neo4j.com/docs/cypher-manual/current/clauses/with/#with-filter-on-aggregate-function-results) for details.

File: src/executor/query/FilterExecutor.cpp [hunk 2]
```diff
@@ -38,7 +38,11 @@ folly::Future<Status> FilterExecutor::execute() {
                                  "the type should be NULL,EMPTY or BOOL");
         }
         if (val.empty() || val.isNull() || !val.getBool()) {
-            iter->unstableErase();
+            if (UNLIKELY(filter->needStableFilter())) {
```
Comment (jievince) [inline src/executor/query/FilterExecutor.cpp:41]: Why use `UNLIKELY`?

Comment (czpmango) [reply src/executor/query/FilterExecutor.cpp:41]: Just provide dynamic branch prediction hints at compile time.

**Review by jievince — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Pull request merged**